nomad/host_volume_endpoint_ce.go

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build !ent
+// +build !ent
```

Comment:
Note for reviewers: https://github.com/hashicorp/nomad-enterprise/commit/5890c50480738bfecb7eca9560218dacea82d690 has my rough prototype for what this will look like in Nomad Enterprise, so the signature here should be what we want.